index.js

```diff
@@ -296,25 +296,36 @@ class CoverageSonar extends CoverageBase {
             const [projectScope, id] = projectKey.split(':');
             const projectName = projectScope === 'pipeline' ? pipelineName : `${pipelineName}:${jobName}`;
             const username = `user-${projectScope}-${id}`;
-
-            return {
+            const projectData = {
                 projectKey,
                 username,
                 projectName,
                 projectScope
             };
+
+            if (projectScope === 'pipeline') {
+                const componentId = encodeURIComponent(`pipeline:${id}`);
+
+                projectData.projectUrl = `${this.sonarHost}/dashboard?id=${componentId}`;
+            }
+
+            return projectData;
         }
 
         // Use user-configured scope; otherwise figure out default scope: pipeline scope for enterprise edition, job scope for everything else
         const userScope = scope && scope !== 'undefined' ? scope : undefined;
         const coverageScope = userScope || (enterpriseEnabled ? 'pipeline' : 'job');
 
         if (coverageScope === 'pipeline') {
```

Comment:
does this not work for job scope? I thought job scope is always job:jobID ?

Reply:
No. Only pipeline scope, we could add more logics to compose the url for job scoped as well. 

That's something I'm not sure if `job:jobID` is unique per pipeline. 

For example, say sd pipeline1 has 2 jobs, we'd like to have a pipeline scoped sonar url on pipeline ui, instead of 2 sonar job urls.